src/sonar_widget/SonarPlot.cc

```diff
@@ -142,3 +148,63 @@ void SonarPlot::rangeChanged(int value)
     range = value;
     drawOverlay();
 }
+
+void SonarPlot::sonarPaletteChanged(int index){
+    colorMapID = (Colormap) index;
+}
+
+void SonarPlot::generateBearingTable(int beams, base::Angle bhoriz, base::Angle bvert) {
+    bearingTable.clear();
+
+    double factor;
+    bhoriz.getDeg() == 120.0 && bvert.getDeg() == 20.0 ? factor = 0.86602540 : factor = 1.0;
+
+    for (int i = 0; i < beams; i++) {
+        double rad = asin(((2 * i - beams) * 1.0 / beams) * factor);
+        const base::Angle new_angle = base::Angle::fromRad(rad);
+        bearingTable.push_back(new_angle);
+
+    }
+}
+
+void SonarPlot::generateColormaps() {
```

Comment:
Yes, definitely. Even OpenCV is fine for me.

The bottom line is: the choice of color is in general very specific to the application (envire MLS goes rainbow because it works there, we picked gradients that match what's customary in imaging sonar visualizations), and the code is really simple. So, in the end, the interface to "configure" the colormap generator will get pretty complicated to account for all of the specificities, or will cover so little than nobody will use it. 

So I really wonder if it makes sense to have a generic piece of code for that purpose.